cmd/main.go

```diff
@@ -51,20 +54,40 @@ func init() {
 	}
 }
 
+// This is a pattern to ensure that deferred functions executes before os.Exit
 func main() {
-	// enable logging
-	klog.InitFlags(nil)
+	os.Exit(run())
+}
+func run() int {
+	// Enable logging in the Kubernetes core package way (support json output)
+	// https://github.com/kubernetes/component-base/tree/master
+	c := logsapi.NewLoggingConfiguration()
+	logsapi.AddGoFlags(c, flag.CommandLine)
 	flag.Parse()
+	logs.InitLogs()
+	if err := logsapi.ValidateAndApply(c, nil); err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		return 1
+	}
 
-	klog.Infof("flags: %v", flag.Args())
+	// Create a context for structured logging, and catch termination signals
+	ctx, cancel := signal.NotifyContext(
+		context.Background(), os.Interrupt, unix.SIGINT)
+	defer cancel()
+	logger := klog.NewKlogr()
```

Comment:
This shouldn't be needed and might never give you a JSON logger.

Leave the context as it is. A `klog.FromContext` will then get the global default prepared by `logsapi.ValidateAndApply`.

```suggestion
	logger := klog.FromContext(ctx)
```